src/Native/Unix/configure.cmake

```diff
@@ -467,17 +467,29 @@ check_cxx_source_compiles(
     HAVE_TCP_VAR_H
 )
 
+check_include_files(
+    sys/cdefs.h
+    HAVE_SYS_CDEFS_H)
+
+if (HAVE_SYS_CDEFS_H)
+    set(CMAKE_REQUIRED_DEFINITIONS "-DHAVE_SYS_CDEFS_H")
+endif()
+
 # If sys/cdefs is not included on Android, this check will fail because
 # __BEGIN_DECLS is not defined
 check_cxx_source_compiles(
     "
+#ifdef HAVE_SYS_CDEFS_H
     #include <sys/cdefs.h>
+#endif
     #include <netinet/tcp.h>
     int main() { int x = TCP_ESTABLISHED; return x; }
     "
     HAVE_TCP_H_TCPSTATE_ENUM
 )
 
+set(CMAKE_REQUIRED_DEFINITIONS)
```

Comment:
Interesting, I was not aware of that function. We should probably be using that in a lot of other places in this file too, like [this section](https://github.com/dotnet/corefx/blob/master/src/Native/Unix/configure.cmake#L336-L385), which is essentially doing a manual set-reset of the variable.